dbt/include/redshift/macros/adapters.sql

```diff
@@ -198,12 +198,57 @@
 
     ),
 
+    data_share as (
+      select
+        a.ordinal_position as columnnum,
+        a.schema_name as schemaname,
+        a.column_name as columnname,
+        case
+          when a.data_type ilike 'character varying%' then
+            'character varying'
+          when a.data_type ilike 'numeric%' then 'numeric'
+          else a.data_type
+        end as col_type,
+        case
+          when a.data_type like 'character%'
+          then nullif(REGEXP_SUBSTR(a.character_maximum_length, '[0-9]+'), '')::int
+          else null
+        end as character_maximum_length,
+        case
+          when a.data_type like 'numeric%' or a.data_type ilike 'integer%'
+          then nullif(
+            SPLIT_PART(REGEXP_SUBSTR(a.numeric_precision, '[0-9,]+'), ',', 1),
+            '')::int
+        end as numeric_precision,
+        case
+          when a.data_type like 'numeric%' or a.data_type ilike 'integer%'
+          then nullif(
+            SPLIT_PART(REGEXP_SUBSTR(a.numeric_scale, '[0-9,]+'), ',', 2),
+            '')::int
+          else null
+        end as numeric_scale
+      from svv_all_columns a
+      inner join (
+        select object_name
+        from svv_datashare_objects
+        where share_type = 'INBOUND'
+        and   object_type in ('table', 'view', 'materialized view', 'late binding view')
+        and   object_name = '{{ relation.schema }}' || '.' || '{{ relation.identifier }}'
+      ) b on a.schema_name || '.' || a.table_name  = b.object_name
+      inner join (
+        select consumer_database from SVV_DATASHARES
+        where share_type = 'INBOUND'
+      ) c on c.consumer_database = a.database_name
+    ),
```

Comment:
my code quality spidey sense tingles at seeing two inner joins of subqueries within a CTE because:
1. inner joins mess with the sequential ordering nature that CTEs provide, and
2. the aliasing of those subqueries statements with `a.` and `b.` is distracting

i highly recommend checking out both our [style guide](https://github.com/dbt-labs/corp/blob/main/dbt_style_guide.md) as well as this follow-on post about [why we love CTEs so much](https://discourse.getdbt.com/t/why-the-fishtown-sql-style-guide-uses-so-many-ctes/1091)